azurerm/internal/services/authorization/parse/role_assignment.go

```diff
@@ -90,6 +99,17 @@ func RoleAssignmentID(input string) (*RoleAssignmentId, error) {
 		}
 		roleAssignmentId.SubscriptionID = id.SubscriptionID
 		roleAssignmentId.ResourceGroup = id.ResourceGroup
+		if id.Provider != "Microsoft.Authorization" && id.Provider != "" {
+			roleAssignmentId.ResourceProvider = id.Provider
+			// logic to save resource scope
+			result := strings.Split(input, "/")
+			for k, v := range result {
+				if v == id.Provider && len(result) >= k+1 {
+					roleAssignmentId.ResourceScope = fmt.Sprintf("%s/%s", result[k+1], result[k+2])
+				}
+			}
```

Comment:
Could it work if the resource scope is complicated? for example: /subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/group1/providers/Microsoft.AppPlatform/Spring/spring1/apps/app1

the expected ResourceScope should be `Spring/spring1/apps/app1`?

Reply:
Fixed, WDYT?